### PR TITLE
Using dockerCmd when possible

### PR DESCRIPTION
**Pull request by fntlnz**

### CODE REVIEW

**Comment by runcom** [comment]
```
These files are not properly gofmt'd:
 - integration-cli/docker_cli_ps_test.go

Please reformat the above files using "gofmt -s -w" and commit the result.
```

File: integration-cli/docker_cli_ps_test.go [hunk 1]
```diff
@@ -256,10 +160,8 @@ func assertContainerList(out string, expected []string) bool {
 
 func (s *DockerSuite) TestPsListContainersSize(c *check.C) {
 
-	cmd := exec.Command(dockerBinary, "run", "-d", "busybox", "echo", "hello")
-	runCommandWithOutput(cmd)
-	cmd = exec.Command(dockerBinary, "ps", "-s", "-n=1")
-	baseOut, _, err := runCommandWithOutput(cmd)
+	exec.Command(dockerBinary, "run", "-d", "busybox", "echo", "hello").Run()
```
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: check error (even if it wasn't)?

File: integration-cli/docker_cli_ps_test.go [hunk 2]
```diff
@@ -651,15 +473,9 @@ func (s *DockerSuite) TestPsLinkedWithNoTrunc(c *check.C) {
 func (s *DockerSuite) TestPsGroupPortRange(c *check.C) {
 
 	portRange := "3800-3900"
-	out, _, err := runCommandWithOutput(exec.Command(dockerBinary, "run", "-d", "--name", "porttest", "-p", portRange+":"+portRange, "busybox", "top"))
-	if err != nil {
-		c.Fatal(out, err)
-	}
+	dockerCmd(c, "run", "-d", "--name", "porttest", "-p", portRange+":"+portRange, "busybox", "top")
```
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: check err (we were here)

File: integration-cli/docker_cli_ps_test.go [hunk 3]
```diff
@@ -651,15 +473,9 @@ func (s *DockerSuite) TestPsLinkedWithNoTrunc(c *check.C) {
...
 
-	out, _, err = runCommandWithOutput(exec.Command(dockerBinary, "ps"))
-	if err != nil {
-		c.Fatal(out, err)
-	}
+	out, _ := dockerCmd(c, "ps")
```
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: here too

File: integration-cli/docker_cli_ps_test.go [hunk 4]
```diff
@@ -540,11 +366,7 @@ func (s *DockerSuite) TestPsListContainersFilterExited(c *check.C) {
 		c.Fatalf("Second in list should be %q, got %q", firstZero, ids[1])
 	}
 
-	runCmd = exec.Command(dockerBinary, "ps", "-a", "-q", "--no-trunc", "--filter=exited=1")
-	out, _, err = runCommandWithOutput(runCmd)
-	if err != nil {
-		c.Fatal(out, err)
-	}
+	out, _ = dockerCmd(c, "ps", "-a", "-q", "--no-trunc", "--filter=exited=1")
```
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: check err

File: integration-cli/docker_cli_ps_test.go [hunk 5]
```diff
@@ -524,11 +354,7 @@ func (s *DockerSuite) TestPsListContainersFilterExited(c *check.C) {
 	}
 
 	// filter containers by exited=0
-	runCmd = exec.Command(dockerBinary, "ps", "-a", "-q", "--no-trunc", "--filter=exited=0")
-	out, _, err := runCommandWithOutput(runCmd)
-	if err != nil {
-		c.Fatal(out, err)
-	}
+	out, _ := dockerCmd(c, "ps", "-a", "-q", "--no-trunc", "--filter=exited=0")
```
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: same

File: integration-cli/docker_cli_ps_test.go (hunk 1 above)
Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: check err (even if we weren't)

**Comment by runcom** [comment]
small nits, rest seems alright

File: integration-cli/docker_cli_events_test.go [hunk 6]
```diff
@@ -40,13 +40,12 @@ func (s *DockerSuite) TestEventsContainerFailStartDie(c *check.C) {
 
 	out, _ := dockerCmd(c, "images", "-q")
 	image := strings.Split(out, "\n")[0]
-	eventsCmd := exec.Command(dockerBinary, "run", "--name", "testeventdie", image, "blerg")
-	_, _, err := runCommandWithOutput(eventsCmd)
+	err := exec.Command(dockerBinary, "run", "--name", "testeventdie", image, "blerg").Run()
```
Comment (runcom) [inline integration-cli/docker_cli_events_test.go]: you could compact the the if below with this statement but it's just sugar

File: integration-cli/docker_cli_ps_test.go (hunk 1 above)
Comment (fntlnz) [inline integration-cli/docker_cli_ps_test.go]: I thought to do so, but I preferred to not change it's behaviour. I'm using dockerCmd now.

Comment (runcom) [inline integration-cli/docker_cli_ps_test.go]: prob ok

**Comment by LK4D4** [comment]
I personally hate `dockerCmd`, so I'm strongly -1 on this. It hides real source of error inside `dockerCmd`.

**Comment by duglin** [comment]
@LK4D4 when you say "real source", do you mean the location of the source file (...test.go) that had the bad cmd?

**Comment by LK4D4** [comment]
@duglin I mean code line too.

**Comment by cpuguy83** [comment]
I like it when the cmd you are running is more of a test setup than the actual thing your testing.
So if it does fail then there is something else majorly wrong.

**Comment by LK4D4** [comment]
I don't like passing `*check.C` anywhere from tests. It is pretty easy to reduce codebase with asserts now without such things.

**Comment by duglin** [comment]
@LK4D4 I was just wondering if it would help to have `dockerCmd` use reflection to figure out the test/line it was called from?   I don't have a strong feeling either way but it is kind of nice to not have to add if-statements/Asserts over and over if we can move 'em all into the util func.

**Comment by LK4D4** [comment]
@duglin I don't feel nice about reading and debugging reflect-code when I can just see where really error was occured. I had problems with that before in integration-cli helpers and I prefer readability over writability and I personally don't care about `if err != nil` or `c.Assert`.

**Comment by runcom** [comment]
I personally don't like it either, shall we close this and maybe create an issue to remove `dockerCmd*` entirely from tests? ping @duglin @LK4D4

**Comment by cpuguy83** [comment]
Actually, go-check does report the correct caller for you.

**Comment by duglin** [comment]
I'm probably in the minority here but I don't have a problem with utility functions doing more than setup work. They are certainly good for that but I also view utility functions as things to make writing testcases easier. So, moving code that appears in a lot of tests into a func to save people from having to duplicate that code over and over is ok with me - even if that func is the one doing the functional testing or doing the c.Fatal() call (as long as it reports the proper info so people know which test to look at to see what was being tested).

But I won't -1 removing it since its not that big of an issue - just more replicated code in the test.

**Comment by runcom** [comment]
alright, I think that now that there is `gocheck`  most of the err checking and fataling is handled with a one liner `c.Assert`. In the end `dockerCmd*` are just pseudo proxies functions that are just for either not checking the error upstream or setting `exec.Command` dir or running with a timeout. I think we should have just one standard way of doing all of this otherwise it could be confusing which way to use (other than personal like). I vote for removing `dockerCmd*` :smile: and have just the `runCommand*` functions handling everything

**Comment by cpuguy83** [comment]
heh, I'd say `runCommand` and `runCommandWithOutput` are much more superfluous, since these are literally just `cmd.Run()` and `cmd.CombinedOutput()`, respectively.

`dockerCmd` actually provides a level of abstraction.
One might even want something like `id := createContainer()`, and remove all theses `strings.TrimSpace`, and custom container names.

**Comment by runcom** [comment]
ok, I just like to have just one way of doing things, above all in tests suite where it's super easy to just implement functions for some use cases and then just forget. Instead a clean interface will make things clear immediately on how to run a container, a container with output etc etc

**Comment by LK4D4** [comment]
@runcom @cpuguy83 Btw I noticed that `gocheck` actually resolves wrong caller problem. So, I'm okay with this if we'll change it to use `c.Assert`

**Comment by runcom** [comment]
@fntlnz could you please rebase and have `dockerCommand` using `c.Assert`?

**Comment by fntlnz** [comment]
Yeah I'll do all this evening along with the other pr assigned to me.

:)

**Comment by fntlnz** [comment]
@runcom @LK4D4 looks good to you?

**Comment by LK4D4** [comment]
LGTM

**Comment by LK4D4** [comment]
ping @runcom

**Comment by calavera** [comment]
LGTM. Merging :tada:

**Pull request merged**